content/en/blog/2021/wasm-api-alpha/index.md

```diff
@@ -0,0 +1,107 @@
+
+The next field below that is the `phase`. This determines where in the proxy’s filter chain the Wasm module will be injected. We have defined four distinct phases for injection:
+
+AUTHN: this is prior to any Istio authentication filters, meaning it is executed even before the remote mTLS certificates are verified.
+AUTHZ: just after the Istio authentication filters and before any authorization filters, i.e. before AuthorizationPolicies have been applied.
+STATS:
+
+`pluginConfig` is used for configuring your Wasm plugin. Whatever you put into this field will be encoded in JSON and passed on to your filter, where you can access it in the configuration callback of the Proxy-Wasm SDKs. For example, you can retrieve the config on `on_configure` in [Rust SDK](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/blob/v0.1.4/src/dispatcher.rs#L255) or `OnPluginStart` call back in [Go SDK](https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/v0.15.0/proxywasm/types/context.go#L74).
+
+The `url` field specifies where to pull the Wasm module. You’ll notice that the `url` in this case is a docker URI - this is because apart from loading Wasm modules via HTTP, HTTPS and the local file system (using file://), we are introducing a container image format to package Wasm extensions for Istio.
+
+## Wasm image specification
+
+We believe that containers are the ideal way to store, publish and manage proxy extensions, so we worked with Solo.io to extend their existing Proxy-Wasm container format with a variant that aims to be compatible with all registries and CLI toolchain. Depending on your processes, you can now either build your proxy extension containers using solo’s wasme tool or straight up with your existing container CLI tooling such as Docker CLI or [buildah](https://buildah.io/). Istio supports both variants.
+
+For detail, please refer to [the link here](https://github.com/solo-io/wasm/tree/master/spec), and learn [how to build OCI images](https://github.com/solo-io/wasm/tree/master/spec#how-can-i-build-images) that are consumable by Istio agent.
+
+## Image fetcher in Istio agent
+
+As of 1.9, Istio-agent provides a reliable solution for loading Wasm binaries fetched from remote HTTP sources configured in the `EnvoyFilters` by leveraging the xDS proxy inside istio-agent and Envoy’s Extension Configuration Discovery Service (ECDS). The same mechanism applies for the new Wasm API implementation in Istio 1.12, and you can use HTTP remote resources reliably without the concern that Envoy might get stuck in bad configurations when the remote fetch fails.
+
+In addition, Istio 1.12 expands this capability to the Wasm OCI images. That means the Istio agent is now able to fetch Wasm images from any OCI registry including Docker Hub, Google Container Registry(GCR), Amazon Elastic Container Registry (Amazon ECR), etc. After fetching images, Istio-agent extracts and caches Wasm binaries from them, and then inserts them into Envoy filter chains.
```

Comment:
Extracts and caches “in memory”

Reply:
I think it is not in-memory, but it actually writes into the local file system https://github.com/istio/istio/blob/master/pkg/wasm/cache.go#L195-L202